lib/inc/internal/facts/linux/os_linux.hpp

```diff
@@ -0,0 +1,29 @@
+/**
+ * @file
+ * Declares the generic Linux operating system query helper.
+ */
+#pragma once
+
+#include <internal/facts/linux/release_file.hpp>
+#include <set>
+#include <map>
+#include <tuple>
+#include <string>
+
+namespace facter { namespace facts { namespace linux {
+
+    struct os_linux
+    {
+        os_linux(std::set<std::string> items = {}, std::string file = release_file::os);
+        virtual std::string get_name(std::string const& distro_id) const;
```

Comment:
I assume passing distro_id is for future use when we refactor the remaining logic (doesn't seem to be used currently)?